experimental/torch_xla2/torch_xla2/ops/op_base.py

```diff
@@ -54,3 +55,16 @@ def wrapper(*args: P.args,
     return wrapper
 
   return decorator
+
+
+def promote_int_input(f: Callable[Concatenate[jax.Array, P], types.JaxValue]):
+   """If the first argument is an int array, promote it to float32."""
+   @functools.wraps(f)
+   def wrapper(x: jax.Array, *args: P.args, **kwargs: P.kwargs):
+      if x.dtype in [jnp.int8, jnp.int16, jnp.int32, jnp.int64]:
+        x = x.astype(jnp.float32)
```

Comment:
Actually it seems to just go to the default dtype:
In [2]: torch.set_default_dtype(torch.float64)

In [3]: torch.sin(torch.tensor(1, dtype=torch.int64))
Out[3]: tensor(0.8415)

In [4]: torch.sin(torch.tensor(1, dtype=torch.int64)).dtype
Out[4]: torch.float64

In [5]: torch.sin(torch.tensor(1, dtype=torch.int32)).dtype
Out[5]: torch.float64

Reply:
Good catch. Fixed